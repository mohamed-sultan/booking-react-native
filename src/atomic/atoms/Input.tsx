import React from 'react';
import {TextInput, StyleSheet} from 'react-native';
import  { useThemeColors } from '@store/useTheme';

const Input = ({value, onChangeText, placeholder, style, ...props}: any) => {
  const colors = useThemeColors();

  const styles = getStyles(colors);
  return (
    <TextInput
      style={[styles.input, style]}
      value={value}
      onChangeText={onChangeText}
      placeholder={placeholder}
      placeholderTextColor={colors.LightText}
      {...props}
    />
  );
};
export default React.memo(Input);

const getStyles = (colors: any) => StyleSheet.create({
  input: {
    paddingVertical: 8,
    fontSize: 16,
    textAlign: 'left',
    color: colors.PrimaryText,
    backgroundColor: colors.LightBackground,
  },
});

