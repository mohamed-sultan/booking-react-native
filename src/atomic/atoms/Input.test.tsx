import React from 'react';
import renderer from 'react-test-renderer';
import Input from './Input';

// Mock the theme hook
jest.mock('@store/useTheme', () => ({
  useThemeColors: () => ({
    PrimaryText: '#000000',
    LightBackground: '#f0f0f0',
    LightText: '#999999',
  }),
}));

describe('Input', () => {
  it('renders correctly with placeholder', () => {
    const tree = renderer
      .create(
        <Input
          value=""
          onChangeText={() => {}}
          placeholder="Enter your name"
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with value and custom style', () => {
    const tree = renderer
      .create(
        <Input
          value="Hello"
          onChangeText={() => {}}
          placeholder="Type here"
          style={{ fontSize: 18, color: 'blue' }}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with additional props (e.g., multiline)', () => {
    const tree = renderer
      .create(
        <Input
          value="Multiline text"
          onChangeText={() => {}}
          placeholder="Type here"
          multiline
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
