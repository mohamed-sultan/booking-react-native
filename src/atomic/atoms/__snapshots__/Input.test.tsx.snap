// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input renders correctly with additional props (e.g., multiline) 1`] = `null`;

exports[`Input renders correctly with placeholder 1`] = `null`;

exports[`Input renders correctly with value and custom style 1`] = `null`;
